qcodes/tests/dataset/test_dataset_basic.py

```diff
@@ -380,7 +285,8 @@ def test_load_by_counter(dataset):
     assert exp.last_counter == 1
 
 
-def test_dataset_with_no_experiment_raises(empty_temp_db):
+@pytest.mark.usefixtures("empty_temp_db")
```

Comment:
do i understand correctly that by using `pytest.mark.usefixtures` we make `mypy` happy, while passing an imported fixture as an argument to `test_*` function makes it complain?

Reply:
I think you can get rid of the unused import warnings from codacy/pylint by adding a ` # pylint: disable=unused-import`  comment to those lines

Reply:
👍 you totes can. It's done in one place only in the PR right now because I'm a lazy dog, but I can add that everywhere so we can hear codacy again.

Reply:
ok, so unused imports over unused arguments :) (just a note: PyCharm since recent version does not complain about an unused argument if that argument is a pytest.fixture ;) )

Reply:
Yes there are two issues.

* Unused imports that codacy complains about. We can either ignore those or mark up the lines 
* Arguments to functions that are seemingly never passed. Neither the decorator nor the name matching is super clean imho but I agree that the decorator is nicer